src/pymodaq_plugins_arduino/hardware/arduino_telemetrix.py

```diff
@@ -33,13 +34,44 @@ def set_pins_output_to(self, value: int):
             self.analog_write(pin, int(value))
         lock.release()
 
+
     def analog_write_and_memorize(self, pin, value):
         lock.acquire()
         value = self.round_value(value)
         self.analog_write(pin, value)
         self.pin_values_output[pin] = value
         lock.release()
 
+    def read_analog_pin(self,data):
+        """
+        Used as a callback function to read the value of the analog inputs.
+        Data[0]: pin_type (not used here)
+        Data[1]: pin_number: i.e. 0 is A0 etc.
+        Data[2]: pin_value: an integer between 0 and 1023 (with an Arduino UNO)
+        Data[3]: raw_time_stamp (not used here)
+        :param data: a list in which are loaded the acquisition parameter analog input
+        :return: a dictionary with the following structure {pin_number(int):pin_value(int)}
+        With an arduino up to 6 analog input might be interrogated at the same time
+        """
+        self.analog_pin_values_input[data[1]] = data[2]*5.0/1023 #Data in Volts
```

Comment:
see here you create it on the fly (wr to my previous remark)

Reply:
the 5 volt encoded on 1024 bits could depend of hte used arduino. Maybe we should include these settings in the config and ask the user to specify the board they use

Reply:
I remove this part for now. The idea is to implement a "calibration" methods i.e. convertin the integers in the expected units: example *5/1023 for volts if ten bytes encoding